### PR TITLE
Multistate

### PR DESCRIPTION
**Pull request by scemama**
I did the changes requested in #40 . I suggest that we merge this branch and close #40

### CODE REVIEW

File: src/trexio_tools/converters/convert_from.py [hunk 1]
```diff
@@ -784,48 +803,63 @@ def run(trexio_filename, filename, filetype, back_end, spin=None, motype=None, s
 
     # Get the basename of the TREXIO file
     try:
-        trexio_basename = os.path.splitext(os.path.basename(trexio_filename))[0]
+        trexio_basename, trexio_extension = os.path.splitext(os.path.basename(trexio_filename))
```
Comment (q-posev) [inline src/trexio_tools/converters/convert_from.py:806]: What if you have `molecule.smth.hdf5`? `trexio_extension` will be `.smth` or `.smth.hdf5` or `.hdf5`?

Comment (scemama) [reply src/trexio_tools/converters/convert_from.py:806]:
```
$ ipython
Python 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:40:32) [GCC 12.3.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.15.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import os

In [2]: trexio_filename = "molecule.smth.hdf5"

In [3]: os.path.splitext(os.path.basename(trexio_filename))
Out[3]: ('molecule.smth', '.hdf5')

```

File: src/trexio_tools/converters/convert_from.py [hunk 2]
```diff
@@ -756,26 +796,70 @@ def run_molden(trexio_file, filename, normalized_basis=True, multiplicity=None,
     trexio.write_mo_coefficient(trexio_file, MoMatrix)
 
 
-def run(trexio_filename, filename, filetype, back_end, spin=None, motype=None):
+def run(trexio_filename, filename, filetype, back_end, spin=None, motype=None, state_suffix=None):
+
+    # Do the cleanup if the file already exists
+    file_cleanup(trexio_filename, back_end)
```
Comment (q-posev) [inline src/trexio_tools/converters/convert_from.py:802]: we now have `remove_trexio_file` which works nicely with the `--overwriten` CLI option. Do we really need to do `file_cleanup` here?

Comment (q-posev) [reply src/trexio_tools/converters/convert_from.py]: Yes, I know how to do that with `argparse` (which is what I use for CLI nowadays) but `docopt` is confusing sometimes. Thanks!

**Pull request merged**